src/lib/models/useDeleteModelMutation.ts

```diff
@@ -1,26 +1,33 @@
-import { useDataEngine } from '@dhis2/app-runtime'
-import { useMutation } from 'react-query'
+import { FetchError, useDataEngine } from '@dhis2/app-runtime'
+import { useMutation, UseMutationOptions } from 'react-query'
+import { ImportSummary } from '../../types'
 import { Schema } from '../useLoadApp'
 
 type MutationFnArgs = {
     id: string
     displayName: string
-    messages?: string[]
 }
 
-type Options = Omit<Parameters<typeof useMutation>[1], 'mutationFn'>
+type DeleteMutationError = Omit<FetchError, 'details'> & {
+    details: ImportSummary
+}
+
+type Options = Omit<
+    UseMutationOptions<ImportSummary, DeleteMutationError, MutationFnArgs>,
+    'mutationFn'
+>
```

Comment:
The `Options` where wrong, mostly due to the inferred types inside `UseMutationOptions`. It's also exported from `react-query`, so it's better to use that. Also helps options passed to have the same inferred types as the ones inside the `useMutation`-declaration